Documentation/ceph-csi-troubleshooting.md

```diff
@@ -441,3 +441,52 @@ $ rbd ls --id=csi-rbd-node -m=10.111.136.166:6789 --key=AQDpIQhg+v83EhAAgLboWIbl
 ```
 
 Where `-m` is one of the mon endpoints and the `--key` is the key used by the CSI driver for accessing the Ceph cluster.
+
+## Node Loss
+
+When a node is lost, you will see application pods on the node stuck in the `Terminating` state while another pod is rescheduled and is in the `ContainerCreating` state.
+
+To allow the application pod to start on another node, force delete the pod.
+
+###  Force deleting the pod
+
+To force delete the pod stuck in the `Terminating` state:
+
+```console
+$ kubectl -n rook-ceph delete pod my-app-69cd495f9b-nl6hf --grace-period 0 --force
+```
+
+After the force delete, wait for a timeout of about 8-10 minutes. If the pod still not in the running state, continue with the next section to blacklist the node.
+
+### Blacklisting a node
+
+To shorten the timeout, you can mark the node as "blacklisted" so Rook can safely failover the pod sooner.
+
+```console
+$ PV_NAME= # enter pv name
+$ IMAGE=$(kubectl get pv $PV_NAME-o jsonpath='{.spec.csi.volumeHandle}' | cut -d '-' -f 6- | awk '{print "csi-vol-"$1}')
+$ echo $IMAGE
+```
+
+The solution is to remove the watcher, following the commands below from the [Rook toolbox](ceph-toolbox.md):
+
+```console
+$ rbd status <image> --pool=<pool name> # get image from above output
+```
+>```
+> Watchers:
+>	watcher=10.130.2.1:0/2076971174 client.14206 cookie=18446462598732840961
+>```
+
+```console
+$ ceph osd blacklist add 10.130.2.1:0 # to know which watcher to block see above output
+blacklisting 10.130.2.1:0
+```
```

Comment:
@travisn before I make changes just to confirm I'll remove the above part and will just ask the user to get node IP(the node which is lost) and blacklist that.

and if Ceph version > octopus we'll use `ceph osd blacklist` else `ceph osd blocklist`

Reply:
If the ceph version is pacific and above use `blocklist` else use `blacklist`.

Reply:
right, thanks

Reply:
Correct, we're just blocking the node, rather than a session id.